ast/commandflag/flags.go

```diff
@@ -138,4 +138,5 @@ type PipelineOpts struct {
 
 type CacheOpts struct {
 	Sharing string `long:"sharing" description:"The cache sharing mode: locked (default), shared, private"`
+	Mode    string `long:"mode" description:"Apply a mode to the cache folder" default:"0644"`
```

Comment:
can we call this `chmod` instead?

I don't see any existing instances of `mode` being used:

```
alex@surfperch:~/gh/earthly/earthly$ grep -RI 'long:"mode"' ast/commandflag/
alex@surfperch:~/gh/earthly/earthly$ grep -RI 'long:"chmod"' ast/commandflag/
ast/commandflag/flags.go:	Chmod           string   `long:"chmod" description:"Apply a mode to the copied files and directories"`
```

Reply:
`RUN --mount`:
![image](https://github.com/earthly/earthly/assets/5653455/419b7b38-825d-4c66-8944-e0f58e925ba2)

Reply:
I also prefer `chmod`, going with it then.

Reply:
ah ha! here's the other instance.

I wonder if `mode` is too generic in the case of `CACHE --mode=...`. I don't think it's obvious that it's a file-permission-mode rather than a mode pertaining to the cache. where as `chmod` is the name of the command line utility for changing file modes, and seems easier to google for rather than just `mode`. 

In the case of the mount option `mode` of the `RUN` command -- it was picked to stay consistent with dockerfiles.

Alternatively maybe we should rename this to `mount-mode` (for both `CACHE` and `COPY` commands) -- that way it would also reflect the `RUN --mount .... mode=...` option. (and deprecate the old names)

@vladaionescu any thoughts on this rename?

Reply:
I was thinking on documenting `chmod` both for `CACHE` and `RUN --mount`, while silently supporting mode for the latter, so dockerfile portability is maintained

Reply:
I like @idelvall's latest suggestion.